pennylane_lightning/lightning_qubit.py

```diff
@@ -133,9 +131,24 @@ def apply_lightning(self, state, operations):
         Returns:
             array[complex]: the output state tensor
         """
-        op_names = [o.name for o in operations]
-        op_wires = [self.wires.indices(o.wires) for o in operations]
-        op_param = [o.parameters for o in operations]
+        op_names = []
+        op_wires = []
+        op_param = []
+
+        for o in operations:
+            op_wires.append(self.wires.indices(o.wires))
+            if o.name in self.kernel_operations:
+                op_names.append(o.name)
+                op_param.append(o.parameters)
+            else:
+                op_names.append("QubitUnitary")
```

Comment:
Here, any gate that does not have an optimized kernel in `lighting.qubit` will be treated as a unitary by the backend. We just need to find the unitary matrix, which we then view as
`[real, imag, real, imag, real, imag...]` using `mat.ravel().view(np.float64)` (no copies required).
These real numbers are then passed as parameters of the gate.

Reply:
Is the issue here that we cannot pass a complex array with pybind and thus need to represent it as a flattened real array?